code/datums/uplink/uplink_items.dm

```diff
@@ -29,6 +29,7 @@ var/datum/uplink/uplink
 	var/name
 	var/desc
 	var/item_cost = 0
+	var/item_limit = 999 // how many times can this item be bought from a uplink (high limit is not shown in the uplink gui)
```

Comment:
consider making this something like where `item_limit = -1` means there is no limit.

Reply:
I'd rather keep it like this. Means the checks and gui handling is simpler, especially when combined with Arrow's request from below.